### PR TITLE
Add `batching` flag to `SQLAlchemyObjectType` and `ORMField`

### PR DESCRIPTION
**Pull request by jnak**
Preview

### CODE REVIEW

File: graphene_sqlalchemy/converter.py [hunk 1]
```diff
@@ -26,29 +33,46 @@ def is_column_nullable(column):
...
+    """
     def dynamic_type():
-        _type = registry.get_type_for_model(model)
+        direction = relationship_prop.direction
+        model = relationship_prop.mapper.entity
+        type_ = obj_type._meta.registry.get_type_for_model(model)
 
-        if not _type:
+        batching_ = batching if is_selectin_available else False
+        connection_field_factory_ = connection_field_factory
```
Comment (Nabellaleen) [inline graphene_sqlalchemy/converter.py:53]: Why do you need this allocation ?

File: graphene_sqlalchemy/converter.py [hunk 2]
```diff
@@ -26,29 +33,46 @@ def is_column_nullable(column):
...
-        elif direction in (interfaces.ONETOMANY, interfaces.MANYTOMANY):
-            if _type._meta.connection:
-                # TODO Add a way to override connection_field_factory
-                return connection_field_factory(relationship_prop, registry, **field_kwargs)
-            return Field(
-                List(_type),
-                **field_kwargs
-            )
+            resolver = get_custom_resolver(obj_type, orm_field_name)
+            if resolver is None:
```
Comment (Nabellaleen) [inline graphene_sqlalchemy/converter.py:60]: I don't find this syntax very readable. maybe something like that ?
```suggestion
            if resolver is None and batching_: 
```

File: graphene_sqlalchemy/converter.py [hunk 3]
```diff
@@ -26,29 +33,46 @@ def is_column_nullable(column):
...
+                resolver = get_batch_resolver(relationship_prop) if batching_ else \
```
Comment (Nabellaleen) [inline graphene_sqlalchemy/converter.py:61]:
```suggestion
                resolver = get_batch_resolver(relationship_prop)
```

File: graphene_sqlalchemy/converter.py [hunk 4]
```diff
@@ -26,29 +33,46 @@ def is_column_nullable(column):
...
+                    get_attr_resolver(obj_type, relationship_prop.key)
```
Comment (Nabellaleen) [inline graphene_sqlalchemy/converter.py:62]:
```suggestion
            elif resolver is None:
                resolver = get_attr_resolver(obj_type, relationship_prop.key) 
```

File: graphene_sqlalchemy/converter.py [hunk 5]
```diff
@@ -26,29 +33,46 @@ def is_column_nullable(column):
...
+
+            return Field(type_, resolver=resolver, **field_kwargs)
+
+        if direction in (interfaces.ONETOMANY, interfaces.MANYTOMANY):
+            if not type_._meta.connection:
+                return Field(List(type_), **field_kwargs)
+
+            if connection_field_factory_ is None:
+                connection_field_factory_ = BatchSQLAlchemyConnectionField.from_relationship if batching_ else \
+                    default_connection_field_factory
```
Comment (Nabellaleen) [inline graphene_sqlalchemy/converter.py:72]: And same thing here, I think :)

File: graphene_sqlalchemy/types.py [hunk 6]
```diff
@@ -152,40 +155,24 @@ def construct_fields(
...
+            field = convert_sqlalchemy_composite(attr, registry, resolver)
         elif isinstance(attr, hybrid_property):
-            field = convert_sqlalchemy_hybrid_method(
-                attr,
-                custom_resolver or _get_attr_resolver(obj_type, orm_field_name, attr_name),
-                **orm_field.kwargs
-            )
+            field = convert_sqlalchemy_hybrid_method(attr, resolver, **orm_field.kwargs)
         else:
             raise Exception('Property type is not supported')  # Should never happen
```
Comment (Nabellaleen) [inline graphene_sqlalchemy/types.py:177]: Very nice cleanup ;)

**Comment by jnak** [comment]
@Nabellaleen Thanks for the review. Unfortunately, I have to close this one and send a new one in the `graphql-python/graphene-sqlalchemy` repo (vs my own) now that the benchmark PR is merged. I'll make sure to address your comments in the new PR.

**Pull request closed without merging**